File: src/api/fca/FCARequest.cs
using System;
using System.Collections.Generic;
using NetTopologySuite.Geometries;
using DVAN.Population;
using DVAN.Accessibility;
using DVAN.Routing;

namespace DVAN.API
{
    /// <summary>
    /// FCA Request.
    /// </summary>
    public class FCARequest
    {
        /// <summary>
        /// Ranges (in sec) to be used by calculation.
        /// </summary>
        /// <example>[180, 360, 540, 720, 900]</example>
        public List<double>? ranges { get; set; }

        public PopulationRequestParams population { get; set; }

        public DecayRequestParams distance_decay { get; set; }

        public RoutingRequestParams? routing { get; set; }

        /// <summary>
        /// Facility Locations in geographic coordinates.
        /// </summary>
        /// <example>[[9.11, 51.23], [9.35, 50.98], [10.02, 52.10]]</example>
        public double[][]? facility_locations { get; set; }

        /// <summary>
        /// Facility capacities.
        /// </summary>
        /// <example>[12.1, 43.5, 21.4]</example>
        public double[]? facility_capacities { get; set; }

        /// <summary>
        /// Calculation mode (one of "isochrones", "isoraster", "matrix").
        /// Defaults to "isochrones".
        /// </summary>
        /// <example>isochrones</example>
        public string? mode { get; set; }
    }
}

File: src/main/java/org/tud/oas/api/accessibility/fca/FCARequest.java
package org.tud.oas.api.accessibility.fca;

import org.tud.oas.requests.AccessResponseParams;
import org.tud.oas.requests.DecayRequestParams;
import org.tud.oas.requests.DemandRequestParams;
import org.tud.oas.requests.RoutingRequestParams;
import org.tud.oas.requests.SupplyRequestParams;
import org.locationtech.jts.geom.Coordinate;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;

@Schema(name = "FCARequest", description = """
                FCA request.
                """)
class FCARequest {
        @JsonProperty("demand")
        public DemandRequestParams demand;

        @JsonProperty("distance_decay")
        public DecayRequestParams distance_decay;

        @JsonProperty("routing")
        public RoutingRequestParams routing;

        @JsonProperty("supply")
        public SupplyRequestParams supply;

        @Schema(name = "mode", description = """
                        Calculation mode (one of "isochrones", "isoraster", "matrix").
                        Defaults to "isochrones".
                        """, example = "isochrones")
        @JsonProperty("mode")
        public String mode;

        @JsonProperty("response")
        public AccessResponseParams response_params;
}
